plugins/search/providers/sphinx_search/scripts/setSphinxLogServerPopulateActive.php

```diff
@@ -0,0 +1,150 @@
+if (!is_numeric($argv[1]) || !in_array($argv[1], array(0, 1)))
+{
+	echo "Error: parameter must be either 0 (disable) or 1 (enable)" . PHP_EOL;
+	exit(1);
+}
+
+define('ROOT_DIR', realpath(dirname(__FILE__) . '/../../../../../'));
+require_once(ROOT_DIR . '/infra/KAutoloader.php');
+require_once(ROOT_DIR . '/alpha/config/kConf.php');
+
+KAutoloader::addClassPath(KAutoloader::buildPath(KALTURA_ROOT_PATH, "vendor", "propel", "*"));
+KAutoloader::addClassPath(KAutoloader::buildPath(KALTURA_ROOT_PATH, "plugins", "*"));
+KAutoloader::setClassMapFilePath(kConf::get("cache_root_path") . '/sphinx/' . basename(__FILE__) . '.cache');
+KAutoloader::register();
+
+error_reporting(E_ALL);
+KalturaLog::setLogger(new KalturaStdoutLogger());
+
+$populateActiveValue = intval($argv[1]);
+$hostname = $argv[2] ?? ($_SERVER["HOSTNAME"] ?? gethostname());
+
+$config = kConf::get('sphinxPopulateSettings', 'sphinx_populate', array());
+if (empty($config))
+{
+	$configFile = ROOT_DIR . "/configurations/sphinx/populate/$hostname.ini";
+	if(!file_exists($configFile))
+	{
+		KalturaLog::err("Configuration file [$configFile] not found.");
+		exit(1);
+	}
+	$config = parse_ini_file($configFile);
+}
+
+$sphinxServer = $config['sphinxServer'] ?? $hostname;
+
+KalturaLog::log("Starting to set [$sphinxServer] populate_active to [$populateActiveValue]");
+
+$dbConf = kConf::getDB();
+DbManager::setConfig($dbConf);
+DbManager::initialize();
+
+$sphinxReadConn = myDbHelper::getConnection(myDbHelper::DB_HELPER_CONN_SPHINX_LOG_READ);
+$sphinxWriteConn = myDbHelper::getConnection(myDbHelper::DB_HELPER_CONN_SPHINX_LOG);
+$sphinxLogServers = SphinxLogServerPeer::retrieveByServer($sphinxServer, $sphinxReadConn);
+
+if (!count($sphinxLogServers))
+{
+	KalturaLog::err("Error: could not find sphinx_log_server records for server hostname [$sphinxServer]");
+	exit(1);
+}
+
+KalturaLog::log("Updating all sphinx_log_server.server [$sphinxServer] sphinx_log_server.populate_active to [$populateActiveValue]");
+
+$lastLogId = array();
+$keepChecking = true;
+$retry = 0;
+$retryAttempts = 3;
+$sleepTime = 5;
+
+// save new value
+foreach ($sphinxLogServers as $sphinxLogServer)
+{
+	/* @var SphinxLogServer $sphinxLogServer */
+	// set populate_active new value
+	$sphinxLogServer->setPopulateActive($populateActiveValue);
+	
+	try
+	{
+		$sphinxLogServer->save($sphinxWriteConn);
+	}
+	catch (Exception $e)
+	{
+		KalturaLog::err("Error trying to save sphinx_log_server, error: [" . $e->getMessage() . "]");
+		exit(1);
+	}
+}
+
+KalturaLog::log("Successfully updated all sphinx_log_server.server [$sphinxServer] sphinx_log_server.populate_active to [$populateActiveValue]");
+KalturaLog::log("Sleeping for 5 seconds to wait for 'populateFromLog.php' to identify the change to sphinx_log_server for [$sphinxServer]");
+sleep(5);
+
+// verify last_log_id progress or not progress
+SphinxLogServerPeer::clearInstancePool();
+$sphinxLogServers = SphinxLogServerPeer::retrieveByServer($sphinxServer, $sphinxReadConn);
```

Comment:
why you need to call it again if you already have it from line 72?

Reply:
I want to get the 'last_log_id' value after the 'populate_active' set to 0 and the 5 sec sleep at line 98
Which should be static now and not progress.

Reply:
@inbal-ben-david I moved the sleep inside the "if ($populateActiveValue === 0)"
since if we set it to 1 there is no need to sleep and wait.